apostrophecms/apostrophe
# Pro 3517 view permission

## Summary

Add support for `viewPermission`

## What are the specific steps to test this change?

1. `npm test`
2. Add `permission` or `viewPermission` to a widget schema and look for warnings in the server console
3. Add `permission` or `viewPermission` to a top level piece type schema field, you should see no warnings
4. Add `permission` or `viewPermission` to a sub level piece type schema field, you should see warnings in the server console
5. On the browser, open the `Network` panel and check the API response request when you open the manager modal for the piece having `permission` or `viewPermission` attributes. Try it with multiple credentials and verify that the fields is visible or not present for each case.

## What kind of change does this PR introduce?
*(Check at least one)*


## Make sure the PR fulfills these requirements:


If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**
